src/webrequest.js

```diff
@@ -223,52 +115,52 @@ function onBeforeRequest(details){
 }
 
 /**
- * Gets the host name for a given tab id
- * @param {Integer} tabId chrome tab id
- * @return {String} the host name for the tab
- */
-function getHostForTab(tabId){
-  var mainFrameIdx = 0;
-  if (!tabData[tabId]) {
-    return;
-  }
-  if (_isTabAnExtension(tabId)) {
-    // If the tab is an extension get the url of the first frame for its implied URL
-    // since the url of frame 0 will be the hash of the extension key
-    mainFrameIdx = Object.keys(tabData[tabId].frames)[1] || 0;
-  }
-  if (!tabData[tabId].frames[mainFrameIdx]) {
-    return;
-  }
-  return extractHostFromURL(tabData[tabId].frames[mainFrameIdx].url);
-}
-
-/**
- * Filters outgoing cookies
+ * Filters outgoing cookies and referer
  * Injects DNT
  *
  * @param details Event details
  * @returns {*} modified headers
  */
 function onBeforeSendHeaders(details) {
-  if (details.tabId == -1){
-    return {};
-  }
-
   if(_isTabChromeInternal(details.tabId)){
     return {};
   }
 
-  var requestAction = checkAction(details.tabId, details.url, false, details.frameId);
-  if (requestAction && Utils.isPrivacyBadgerEnabled(getHostForTab(details.tabId))) {
+  var tabDomain = getHostForTab(details.tabId);
+  var requestDomain = window.extractHostFromURL(details.url);
+
+  if (Utils.isPrivacyBadgerEnabled(tabDomain) && 
+      window.isThirdParty(requestDomain, tabDomain)) {
+    var requestAction = checkAction(details.tabId, details.url, false, details.frameId);
+    // If this might be the third stike against the potential tracker which
+    // would cause it to be blocked we should check immediately if it will be blocked.
+    if (requestAction == pb.ALLOW && 
+        pbStorage.getTrackingCount(requestDomain) == pb.TRACKING_THRESHOLD - 1){
+      pb.heuristicBlocking.heuristicBlockingAccounting(details);
+      requestAction = checkAction(details.tabId, details.url, false, details.frameId);
+    }
+
+    // This will only happen if the above code sets the action for the request
+    // to block
+    if (requestAction == pb.BLOCK) {
+      // Notify the content script...
+      var msg = {
+        replaceSocialWidget: true,
+        trackerDomain: window.extractHostFromURL(details.url)
+      };
+      chrome.tabs.sendMessage(details.tabId, msg);
 
-    if (requestAction == "cookieblock" || requestAction == "usercookieblock") {
```

Comment:
nope! This line was replaced with pb.COOKIEBLOCK and so forth. This shouldn't be a change in when social widget replacement happens.